src/Storages/ProjectionsDescription.cpp

```diff
@@ -348,7 +346,7 @@ String ProjectionsDescription::toString() const
     for (const auto & projection : projections)
         list.children.push_back(projection.definition_ast);
 
-    return serializeAST(list);
+    return list.formatWithSecretsOneLine();
```

Comment:
`toString()` methods require extra attention because secrets may still leak into error messages (or logs), e.g: in `ReplicatedMergeTreeTableMetadata.cpp`:
```
    String parsed_zk_projections = ProjectionsDescription::parse(from_zk.projections, columns, context).toString();
    if (projections != parsed_zk_projections)
    {
        throw Exception(ErrorCodes::METADATA_MISMATCH, "Existing table metadata in ZooKeeper differs in projections. "
                "Stored in ZooKeeper: {}, parsed from ZooKeeper: {}, local: {}",
                from_zk.projections, parsed_zk_projections, projections);
    }    
```

I would suggest creating counterparts for these methods used for logging – `toStringWithoutSecrets` or similar – maybe as a follow-up.